TestLauncher_BucketDelete sometimes fails with a handled panic

I don't have repro steps, and it seems to be pretty rare, but I occasionally see this error:

```
--- FAIL: TestLauncher_BucketDelete (0.41s)
    launcher_test.go:150: unexpected status code: 500, body: {"code":"internal error","message":"a panic has occurred","error":"runtime error: invalid memory address or nil pointer dereference"}, headers: map[Content-Type:[application/json; charset=utf-8] X-Platform-Error-Code:[internal error] Date:[Wed, 16 Jan 2019 19:42:33 GMT] Content-Length:[133]]
FAIL
```

I just ran into this too at https://circleci.com/gh/influxdata/influxdb/21243. Reproduced locally with 

```
$ go test -run=TestLauncher_BucketDelete ./cmd/influxd/launcher/ -count=10
--- FAIL: TestLauncher_BucketDelete (0.44s)
    launcher_test.go:150: unexpected status code: 500, body: {"code":"internal error","message":"a panic has occurred","error":"runtime error: invalid memory address or nil pointer dereference"}, headers: map[Content-Type:[application/json; charset=utf-8] X-Platform-Error-Code:[internal error] Date:[Thu, 17 Jan 2019 22:02:45 GMT] Content-Length:[133]]
FAIL
FAIL	github.com/influxdata/influxdb/cmd/influxd/launcher	4.965s
```

Probably out of scope of this issue, but it would be nice to include the panic details in the test output, if possible. I couldn't get at the stack trace even under `go test -v`.

--- a/tsdb/tsm1/engine_delete_measurement.go
+++ b/tsdb/tsm1/engine_delete_measurement.go
@@ -149,6 +149,9 @@ func (e *Engine) DeleteBucket(name []byte, min, max int64) error {
 
 		// In this case the entire measurement (bucket) can be removed from the index.
 		if min == math.MinInt64 && max == math.MaxInt64 {
+			// The TSI index and Series File do not store series data in escaped form.
+			name = models.UnescapeMeasurement(name)
+
 			// Build up a set of series IDs that we need to remove from the series file.
 			set := tsdb.NewSeriesIDSet()
 			itr, err := e.index.MeasurementSeriesIDIterator(name)

